examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -115,28 +231,88 @@ int main()
+        attachment.loadOp = nbl::video::IGPURenderpass::ELO_CLEAR;
+        attachment.storeOp = nbl::video::IGPURenderpass::ESO_STORE;
+        
+        nbl::video::IGPURenderpass::SCreationParams::SSubpassDescription::SAttachmentRef colorAttachmentRef;
+        colorAttachmentRef.attachment = 0u;
+        colorAttachmentRef.layout = nbl::asset::EIL_UNDEFINED;
+
+        nbl::video::IGPURenderpass::SCreationParams::SSubpassDescription subpassDescription;
+        subpassDescription.colorAttachmentCount = 1u;
+        subpassDescription.colorAttachments = &colorAttachmentRef;
+        subpassDescription.depthStencilAttachment = nullptr;
+        subpassDescription.flags = nbl::video::IGPURenderpass::ESDF_NONE;
+        subpassDescription.inputAttachmentCount = 0u;
+        subpassDescription.inputAttachments = nullptr;
+        subpassDescription.preserveAttachmentCount = 0u;
+        subpassDescription.preserveAttachments = nullptr;
+        subpassDescription.resolveAttachments = nullptr;
+
+        nbl::video::IGPURenderpass::SCreationParams renderpassParams;
+        renderpassParams.attachmentCount = 1u;
+        renderpassParams.attachments = &attachment;
+        renderpassParams.dependencies = nullptr;
+        renderpassParams.dependencyCount = 0u;
+        renderpassParams.subpasses = &subpassDescription;
+        renderpassParams.subpassCount = 1u;
+
+        auto gpuRenderpass = logicalDevice->createGPURenderpass(renderpassParams);
+
+        nbl::video::IGPUGraphicsPipeline::SCreationParams graphicsPipelineParams;
+        graphicsPipelineParams.renderpassIndependent = core::smart_refctd_ptr<nbl::video::IGPURenderpassIndependentPipeline>(const_cast<video::IGPURenderpassIndependentPipeline*>(gpumesh->getMeshBuffers().begin()[0]->getPipeline()));
+        graphicsPipelineParams.renderpass = gpuRenderpass;
+
+        gpuGraphicsPipeline = logicalDevice->createGPUGraphicsPipeline(nullptr, std::move(graphicsPipelineParams));
+    }
+
+	scene::ICameraSceneNode* camera = sceneManager->addCameraSceneNodeFPS(0,100.0f,0.5f);
 
 	camera->setPosition(core::vector3df(-4,0,0));
 	camera->setTarget(core::vector3df(0,0,0));
 	camera->setNearValue(1.f);
 	camera->setFarValue(5000.0f);
 
-    smgr->setActiveCamera(camera);
+    sceneManager->setActiveCamera(camera);
+
+    nbl::asset::SViewport viewport;
+    {
+        viewport.maxDepth = 0.f;
+        viewport.minDepth = 1.f;
+        viewport.x = 0u;
+        viewport.y = 0u;
+        viewport.width = WIN_W;
+        viewport.height = WIN_H;
+    }
 
-	uint64_t lastFPSTime = 0;
-	while(device->run() && receiver.keepOpen())
+	while(true)
 	{
-		driver->beginScene(true, true, video::SColor(255,255,255,255) );
+        // Shouldn't I need to invoke something like beginScene and endScene? Is there any such call on new API?
+
+        commandBuffer[0]->setViewport(0, 1, &viewport);
+
+        const size_t msTimeCount = 69; // device->getTimer()->getTime()).count(), how to get time in new API properly?
```

Comment:
what about this one?

Reply:
no api for this right now afaik

Reply:
no need for it, because there's no implicit `OnAnimate` for the camera